lib/home/movie_service.dart

```diff
@@ -1,34 +1,46 @@
 import 'package:dio/dio.dart';
-import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_command/flutter_command.dart';
+import 'package:get_it/get_it.dart';
 import 'package:movieapp/environment_config.dart';
 import 'package:movieapp/home/movies_exception.dart';
 
 import 'movie.dart';
 
-final movieServiceProvider = Provider<MovieService>((ref) {
-  final config = ref.watch(environmentConfigProvider);
-
-  return MovieService(config, Dio());
-});
 
 class MovieService {
-  MovieService(this._environmentConfig, this._dio);
+  MovieService([this._environmentConfig, this._dio]) {
+    _environmentConfig ??= GetIt.I<EnvironmentConfig>();
+    _dio ??= Dio();
+  }
 
-  final EnvironmentConfig _environmentConfig;
-  final Dio _dio;
+  EnvironmentConfig _environmentConfig;
+  Dio _dio;
```

Comment:
Is it possible to pass the objects through the constructor instead to keep them final?

This would also minimize tha change :)

Reply:
Nope, because I want to show a pattern if you use GetIt here. I really hate it that you can't assign a value one time in a constructor

Reply:
Have not tried it out, is this a limitation or just something you prefer doing with Get_it?

Reply:
Most of the time I don't assign objects inside GetIt to final properties. I just didn't want to change too much. 
Normally I access always directly over GetIt or use a short lived final local variable if I have to make many accesses to the same object.

Reply:
I just learned we can make this variable soon ´late final´